src/prefect/server/models/task_runs.py

```diff
@@ -428,8 +428,8 @@ async def set_task_run_state(
     proposed_state_type = state.type if state else None
     intended_transition = (initial_state_type, proposed_state_type)
 
-    if run.flow_run_id is None:
-        task_policy = AutonomousTaskPolicy  # CoreTaskPolicy + prevent `Running` -> `Running` transition
+    if state.state_details.deferred:
+        task_policy = BackgroundTaskPolicy  # CoreTaskPolicy + prevent `Running` -> `Running` transition
```

Comment:
We should expand the BackgroundTaskPolicy to explain why we prevent Running -> Running transitions on background tasks. The reason is that the design allows people to run multiple task servers, and servers might see the same task runs. We use optimistic concurrency control here -- letting servers see the same runs and then ignoring attempts to run already-running tasks.

Reply:
Doesn't have to be this PR though. I can add that separately.